api/connector/simpleconnector_test.go

```diff
@@ -100,3 +102,34 @@ func (s *simpleConnectorSuite) TestNewSimpleRespectsClientCredentials(c *gc.C) {
 		}
 	}
 }
+
+func (s *simpleConnectorSuite) TestSimpleConnectorConnect(c *gc.C) {
+	connector, err := NewSimple(SimpleConfig{
+		Username:            "alice@canonical.com",
+		ControllerAddresses: []string{"localhost:17080"},
+	})
+	c.Assert(err, gc.IsNil)
+
+	var called bool
+
+	openFunc := func(i *api.Info, do api.DialOpts) (api.Connection, error) {
+		called = true
+
+		// Zeros to false, ensure it is true after Connect dial opt.
+		c.Assert(do.InsecureSkipVerify, gc.Equals, true)
+
+		// Defaults to 10 * time.Minute, ensure it is overwritten after Connect dial opt.
+		c.Assert(do.Timeout, gc.Equals, 5*time.Minute)
+		return nil, nil
+	}
+
+	connector.Connect(
+		openFunc,
+		func(do *api.DialOpts) {
+			do.InsecureSkipVerify = true
```

Comment:
just do 
```
called = true
```
here.. then we don't need openFunc

Reply:
We can't check the values are actually changed if we remove it. We just assert it's called, and the previous issue wouldn't have been tested. @SimonRichardson thoughts?